### PR TITLE
Support hapi 17

### LINKED ISSUE #36: Deprecate?
If not, please open a hapi v17 migration issue.

### LINKED ISSUE #37: Migrate to hapi 17

### PR DESCRIPTION
**Pull request by geek**
Closes #37 
Closes #36

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -3,8 +3,9 @@
 // Load modules
 
 const Net = require('net');
-const Hoek = require('hoek');
 const Repl = require('repl');
+const Barrier = require('cb-barrier');
```
Comment (cjihrig) [inline lib/index.js]: Nice

File: lib/index.js [hunk 2]
```diff
@@ -18,22 +19,24 @@ const internals = {
...
 
+    const barrier = new Barrier();
     const tcp = Net.createServer((socket) => {
 
         if (settings.localOnly) {
             const address = socket.address();
             if ((address.family === 'IPv6' && address.address !== '::1' && address.address !== '::ffff:127.0.0.1')
-            || (address.family === 'IPv4' && address.address !== '127.0.0.1')) {
+                || (address.family === 'IPv4' && address.address !== '127.0.0.1')) {
+
```
Comment (cjihrig) [inline lib/index.js]: Blank line?

File: test/index.js [hunk 3]
```diff
@@ -21,431 +22,424 @@ const it = lab.it;
...
+    const server = Hapi.server();
+    const port = await internals.availablePort();
 
-            expect(err).to.not.exist();
+    await server.register({ plugin: Reptile, options: { port } });
 
-            const address = Net.Socket.prototype.address;
-            Net.Socket.prototype.address = function () {
+    const address = Net.Socket.prototype.address;
+    Net.Socket.prototype.address = function () {
```
Comment (cjihrig) [inline test/index.js]: Consider making a function to monkey patch this in the future since it is done a few times.

File: package.json [hunk 4]
```diff
@@ -13,15 +13,13 @@
     "node": ">=4"
```
Comment (Marsup) [inline package.json]: Missing the engine bump.

**Pull request merged**